server/src/main/java/org/elasticsearch/cluster/ClusterFeatures.java

```diff
@@ -244,15 +244,14 @@ public ClusterFeatures apply(ClusterFeatures part) {
 
     @Override
     public Iterator<? extends ToXContent> toXContentChunked(ToXContent.Params params) {
-        return Iterators.concat(
-            Iterators.single((builder, p) -> builder.startArray()),
-            nodeFeatures.entrySet().stream().sorted(Map.Entry.comparingByKey()).<ToXContent>map(e -> (builder, p) -> {
+        return ChunkedToXContentBuilder.builder(params)
+            .startArray()
+            .forEach(nodeFeatures.entrySet().stream().sorted(Map.Entry.comparingByKey()).iterator(), (e, b) -> {
                 String[] features = e.getValue().toArray(String[]::new);
                 Arrays.sort(features);
-                return builder.startObject().field("node_id", e.getKey()).array("features", features).endObject();
-            }).iterator(),
-            Iterators.single((builder, p) -> builder.endArray())
-        );
+                b.startObject().field("node_id", e.getKey()).array("features", features).endObject();
```

Comment:
This does create a lot more chunks. I'm not sure this matters given the small scales involved?